### ISSUE TITLE
svg on Safari 7.0.3 (OSX 10.9.2) 

### ISSUE THREAD

**Issue by pw374** [ocaml/v2.ocaml.org#469]
I've just noticed that some SVG files are displayed a little too black...
cf. the 2 rounded-corner black squares on this screenshot:
![ocaml-org](https://cloud.githubusercontent.com/assets/3776012/2841462/6e2ab4fe-d067-11e3-8d99-ba25782ad577.jpg)

It's very strange since Safari displays the images correctly when opened alone (i.e., not embedded in a page).

**Comment by dbuenzli** [comment]
The svg seem to be a little bit messy. First if you open the .svg files in inkscape you see that each svg file contains _all_ the icons, only the cropbox is moved, some bandwidth could be saved here. Then the white is not simply a white path over a gray background it seems to be produced by some kind of clipping and compositing (didn't look in details). In any case with svg, it's better to stick to simple way of producing images (in this case a filled white path over gray rounded box seems appropriate), it's an insane standard and hard to fully implement correctly. Simple is also better for mobile support/performance.

**Comment by Chris00** [comment]
On Wed, 30 Apr 2014 06:38:42 -0700, Daniel Bünzli wrote:

> The svg seem to be a little bit messy. First if you open the .svg
> files in inkscape you see that each svg file contains all the icons,
> only the cropbox is moved, some bandwidth could be saved here.

I did that for some but had not the courage to do it for all of them.

> Then the white is not simply a white path over a gray background it
> seems to be produced by some kind of clipping and compositing
> (didn't look in details).

I suspect the design firm use the Adobe suite and has no idea about
these "details" either.

> In any case with svg, it's better to stick to simple way of
> producing images (in this case a filled white path over gray rounded
> box seems appropriate), it's an insane standard and hard to fully
> implement correctly. Simple is also better for mobile support/performance.

Doesn't this amount to redo the logos?  Do not know what Amir may do
with the firm...

**Comment by Chris00** [comment]
@pw374 I use the latest Firefox and have no problem.  This is annoying if there are problems on some platforms.  @amirmc Was the firm supposed to test the compatibility with existing browsers?

**Comment by pw374** [comment]
I don’t have the problem with Firefox (28.0, not the latest) either.

I haven’t looked into the SVG files, but Daniel’s comments make a lot of sense.
The only way it makes sense to have a “big" SVG file with multiple images inside is when a script dynamically selects the portion to display, in which case there’s only one file transfer for many pictures. But it our case it’s nonsense since we have a specific file for each pictogram.

**Comment by johnwhitington** [comment]
This is still true in Safari 8.0.6 as of today. I will have a look at the file and see what I can do.

**Comment by Chris00** [comment]
@johnwhitington Thanks.

**Comment by johnwhitington** [comment]
announcement.svg also fails (see top news item at the moment).

Until this can be fixed properly, is there a way the site can be configured to serve the substitute PNGs to Safari even though it claims to support SVG? Safari isn't widely used on the desktop (3.8%) but of course it also fails on iOS, where almost everyone uses Safari...

Short of importing them into a graphics package and fiddling with them and re-exporting, I don't see an easy solution. Do we have the illustrator sources, by the way?

**Comment by Chris00** [comment]
@johnwhitington May you force-reload and tell me?  (I simplified the SVGs a lot but I do not know the cause of the problem for Safari.)

**Comment by johnwhitington** [comment]
No change here, I'm afraid, after a full reload.

**Comment by amirmc** [comment]
> Do we have the illustrator sources, by the way?

We only have the files you see in https://github.com/ocaml/ocaml-logo. Specifically, we have EPS files.

**Comment by Chris00** [comment]
I tried several variations to no avail (using some services online to display the page with Safari — this is a pain).  It seems that, when one [displays the image by itself](https://ocaml.org/img/community-large.svg) and go back to the page, everything is fine.  Crazy, huh!

If you change the community image by another one, does the same problem appears?

**Comment by agarwal** [comment]
> It seems that, when one displays the image by itself and go back to the page, everything is fine. Crazy, huh!

I can't reproduce that. The image by itself does show fine, but never shows on the frontpage.

**Comment by Chris00** [comment]
What about using another SVG image at that location?  Does HTML need a “fix” and not the SVG???

**Comment by agarwal** [comment]
> What about using another SVG image at that location?

I replaced `src="img/community-large.svg"` to `src="img/platform-large.svg"`, and then the image does show up fine.

I opened all 4 svg files in Illustrator, and I notice that learn-large.svg, documentation-large.svg, and platform-large.svg all contain all of the images within them. However, community-large.svg contains only the community icon. So at least there is this one difference.

Question: why are both svg and png files included in the html? Removing the png images seems to have no effect.

**Comment by Chris00** [comment]
On Wed, 10 Jun 2015 07:15:49 -0700, Ashish Agarwal wrote:

> Question: why are both svg and png files included in the html? Removing the png images seems to have no effect.

The CSS "display:" is set according to the capabilities of the browser.

**Comment by johnwhitington** [comment]
See pull request #669 for a new SVG file for community-large.svg

**Comment by agarwal** [comment]
@johnwhitington Thanks! Want to educate us on what the issue was?

**Comment by agarwal** [comment]
The change is live, and working on Safari. Closing this issue!

**Comment by johnwhitington** [comment]
Instead of just being a gray rounded rectangle with some white shapes on top, the SVG seemed to be a rectangle clipped using a clipping path representing the substantive parts of the logo. That is to say, if it had a trasparent background, it would have holes in it. Nothing wrong with that, in theory.

There was some other stuff to do with transparency that I didn't really understand, so I just stripped it out. Really, the logos are just a few paths.

When I have time, I'll check through the whole website to make sure there are no more problems, but we have fixed the most obvious ones.
